go/cmd/dolt/commands/revert.go

```diff
@@ -79,89 +85,66 @@ func (cmd RevertCmd) Exec(ctx context.Context, commandStr string, args []string,
 		return 1
 	}
 
-	if dEnv.IsLocked() {
-		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(env.ErrActiveServerLock.New(dEnv.LockFile())), help)
-	}
-
-	_, sqlCtx, closer, err := cliCtx.QueryEngine(ctx)
+	queryist, sqlCtx, closeFunc, err := cliCtx.QueryEngine(ctx)
 	if err != nil {
-		cli.PrintErrln(err.Error())
+		cli.Println(err.Error())
 		return 1
 	}
-	defer closer()
-
-	headCommit, err := dEnv.HeadCommit(sqlCtx)
-	if err != nil {
-		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
-	}
-	headRoot, err := headCommit.GetRootValue(sqlCtx)
-	if err != nil {
-		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
-	}
-	workingRoot, err := dEnv.WorkingRoot(sqlCtx)
-	if err != nil {
-		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
-	}
-	headHash, err := headRoot.HashOf()
-	if err != nil {
-		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
-	}
-	workingHash, err := workingRoot.HashOf()
-	if err != nil {
-		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
-	}
-	if !headHash.Equal(workingHash) {
-		cli.PrintErrln("You must commit any changes before using revert.")
-		return 1
+	if closeFunc != nil {
+		defer closeFunc()
 	}
 
-	headRef := dEnv.RepoState.CWBHeadRef()
-	commits := make([]*doltdb.Commit, apr.NArg())
-	for i, arg := range apr.Args {
-		commitSpec, err := doltdb.NewCommitSpec(arg)
-		if err != nil {
-			return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
-		}
-		commit, err := dEnv.DoltDB.Resolve(sqlCtx, commitSpec, headRef)
+	var author string
+	if apr.Contains(cli.AuthorParam) {
+		author, _ = apr.GetValue(cli.AuthorParam)
+	} else {
+		name, email, err := env.GetNameAndEmail(cliCtx.Config())
 		if err != nil {
-			return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+			cli.Println(err.Error())
+			return 1
 		}
-		commits[i] = commit
+		author = name + " <" + email + ">"
 	}
-	tmpDir, err := dEnv.TempTableFilesDir()
-	if err != nil {
-		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+
+	var params []interface{}
+	params = append(params, author)
+
+	var buffer bytes.Buffer
+	buffer.WriteString("CALL DOLT_REVERT('--author', ?")
+	// Loop over args and add them to the query
+	for _, input := range apr.Args {
+		buffer.WriteString(", ?")
+		params = append(params, input)
 	}
-	opts := editor.Options{Deaf: dEnv.DbEaFactory(), Tempdir: tmpDir}
-	workingRoot, revertMessage, err := merge.Revert(sqlCtx, dEnv.DoltDB, workingRoot, headCommit, commits, opts)
+	buffer.WriteString(")")
+	query, err := dbr.InterpolateForDialect(buffer.String(), params, dialect.MySQL)
 	if err != nil {
-		return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+		cli.Println(err.Error())
```

Comment:
Should we be including other error information in these 3 cases? (Lines 122, 128, 133.) For example, on line 139, we're calling out that the main operation succeeded, but something else failed.
These error can be improved a bit by being clear about what failed: interpolation on line 122, query on line 128, reading the results on 131.

Reply:
In the first and third cases, I don't expect us to every hit those in normal usage. I did update the second to state the sql query which failed. Will change the output a little, but should be ok to pass tests.